src/models/disassemblymodel.cpp

```diff
@@ -40,6 +51,18 @@ void DisassemblyModel::setDisassembly(const DisassemblyOutput& disassemblyOutput
 {
     beginResetModel();
     m_data = disassemblyOutput;
+
+    QString sourceCode;
```

Comment:
feels ugly to join the lines just to let them be split up again in the document again. afair you can manually insert individual lines too via the QTextCursor API (a bit hidden but would be cool if you could have a look)

something like

```
    QTextCursor cursor(m_document);
    cursor.insertText(line);
```

potentially also `cursor.insertBlock` to split lines, unsure if that's needed